### PR TITLE
BioSequences v3 and BioSymbols v5

### PR DESCRIPTION
**Pull request by CiaranOMara**
This PR:
- increments dependency compatibilities (reimplements a reinterpret of BioSymbols);
- includes some doctests in the unit tests; and
- updates doctests for BioSequences v3 and BioSymbols v5 (adjusts a print width).

### CODE REVIEW

**Comment by jakobnissen** [comment]
@CiaranOMara I don't see a reason to bump BioAlignments version to 3.0.0, AFAIK, this is not a breaking change. If someone has BioSequences v2 installed, this change will not be breaking, because the Pkg resolver will simply not install this version. Conversely, bumping the major version can cause a bunch of issues for downstream packages that rely on BioAlignments but not BioSequences.

Otherwise LGTM

**Review by jakobnissen — CHANGES_REQUESTED** [review]
No need to increment major version number.

**Comment by CiaranOMara** [comment]
The code in this PR doesn't appear to warrant a major version, but since v2 of this package, the method signature for the `AlignmentAnchor` changed, which warrants the major version change or some sort of patch (reported in #69).

**Comment by jakobnissen** [comment]
@CiaranOMara Ah, I see. Maybe it's better to revert the breaking change from 2.0.0, since there seem to not be a release that actually includes the change, now that 2.1.0 has been yanked?

**Comment by CiaranOMara** [comment]
Having had another look to get back up to speed, the breaking change of the adition of the `alnpos` field to the `AlignmentAnchor` struct was merged in #44. My sense is that adding `alnpos` is a useful optimisation, though I've not yet done much in this area. The inclusion of the field is at least important to @alyst.

I'm in favour of moving forward, mainly because #44 was the bulk of the work since v2, but on reflection it would be worth slowing down to address other breaking changes suggested in https://github.com/BioJulia/BioAlignments.jl/pull/44#issuecomment-697350429 before incrementing a major version number.

If moving forward with `alnpos`, I think we'd need a few new outer constructors as well as `push!` and `append!` methods  for `Alignment` to help with usability by calculating `alnpos` as `Operation` and their lengths are added, but these would be features.

I've removed the major version increment.

**Comment by jakobnissen** [comment]
I think this illustrates very well why most people think that fields of structs should be implementation details, and the API instead should only expose accessor functions - but I digress.
You're right - we can have a minor version change for now, then later review what breaking changes we want, if any,

**Review by jakobnissen — APPROVED** [review]

**Comment by CiaranOMara** [comment]
> You're right - we can have a minor version change for now, then later review what breaking changes we want, if any,

We'd need to branch and set a new master before #44 to take/accept minor changes.

**Comment by MillironX** [comment]
Any progress on this? I'm excited for a new release of BioAlignments, but everything seems to be stuck here. I think this needs to be treated as a [hotfix under OneFlow](https://www.endoflineblog.com/oneflow-a-git-branching-model-and-workflow#hotfix-branches).

1. Create a new branch before #44 
    ```bash
    git checkout -b hotfix/2.2.0 118c5ad3
    ```
2. Cherry-pick the commits from the CI/TagBot PRs (so we can register the new version)
    - #50
    - #71
3. Cherry-pick the commits from this PR (#72)
    ```bash
    git remote add ciaranomara git@github.com:CiaranOMara/BioAlignments.jl.git
    git fetch ciaranomara
    git cherry-pick c32e7678
    ...
    ```
4. Do version-bumping/testing stuff
5. Push the `hotfix/2.2.0` branch to GitHub
6. Trigger JuliaRegistrator on the tip of the new branch
7. Once the version is registered and TagBot has done its thing, then merge `hotfix/2.2.0` into `master` and delete
    ```bash
    git merge hotfix/2.2.0 master
    git push origin :hotfix/2.2.0
    ```

I have done steps 1-4 on my own machine, so I know that works as expected, but everything else needs to be done by someone with write access to the repo.

**Comment by kescobo** [comment]
@MillironX Can you not do 1-5 on a fork and then PR?

I'm happy to take these steps, but don't want to do so until one of @CiaranOMara or @jakobnissen has weighed in

**Comment by MillironX** [comment]
> @MillironX Can you not do 1-5 on a fork and then PR?

I can prepare a PR, but it won't have a target (and it won't show up in GitHub) until someone creates the `hotfix` branch on this repo.

**Comment by kescobo** [comment]
Oh, I get it.

**Comment by CiaranOMara** [comment]
@MillironX, that is an excellent suggestion; I have drafted the [hotfix branch](https://github.com/BioJulia/BioAlignments.jl/pull/76) and will invite people to review it in the next few days. 

Since the goal/focus of this PR has changed, it needs tidying up. I'll split this PR, focusing on doctests here and BioSymbols v3 in a new PR. After merging those two PRs into master, we can neatly cherry-pick them into the hotfix.

**Comment by CiaranOMara** [comment]
I think this is ready for review again.

File: .github/workflows/ci.yml
Comment (kescobo) [inline .github/workflows/ci.yml:19]: Oh man, this is long overdue...

**Review by kescobo — APPROVED** [review]
LGTM.

**Pull request merged**
